pkg/skaffold/runner/listen.go

```diff
@@ -39,13 +41,17 @@ type SkaffoldListener struct {
 	intentChan <-chan bool
 }
 
-func (l *SkaffoldListener) LogWatchToUser(out io.Writer) {
+func (l *SkaffoldListener) LogWatchIsActive(out io.Writer) {
 	l.Trigger.LogWatchToUser(out)
 }
 
+func (l *SkaffoldListener) LogWatchIsInactive(out io.Writer) {
+	color.Yellow.Fprintln(out, "Not watching for changes...")
+}
+
 // WatchForChanges listens to a trigger, and when one is received, computes file changes and
-// conditionally runs the dev loop.
-func (l *SkaffoldListener) WatchForChanges(ctx context.Context, out io.Writer, devLoop func() error) error {
+// conditionally runs the dev loop. It logs to user only when the dev loop is active.
+func (l *SkaffoldListener) WatchForChanges(ctx context.Context, out io.Writer, fn func() (isActive bool, devLoop func() error)) error {
```

Comment:
Is it possible to avoid starting the watcher if we're not interested in any changes?  It is weird that we log file changes at a DEBUG level when we're not actually doing anything with them.

(We can still initiate rebuilds with the API.)

Reply:
That just seems to be how it's designed though. The file watcher aggregates all changes since last build or deploy and on the API trigger to rebuild or redeploy it does so intelligently. If I don't start the watcher, then I'll have to redo the whole API also and the end result will be a different code path for debug(no devloop). WDYT?

Reply:
Implemented in #4676